### PR TITLE
Install GHC bindist in relocatable mode on Unix

### PR DESCRIPTION
**Pull request by aherrmann**
On Unix the GHC bindist requires to execute `./configure && make && make install`. In its default configuration, it hard-codes the absolute path to the bindist installation dir into various places in the final bindist. This affects the hashes and thereby cache keys, as the absolute install path will likely differ on different machines. This patch builds the bindist in relocatable mode on Unix and patches the generated wrapper scripts to no longer hard-code the absolute installation path.

- Pin the `--libdir` to `EPREFIX/lib`. On some platforms autotools defaults to `EPREFIX/lib64`, instead.
- Patch the configure script to build in relocatable mode.
- Patch the generated executable wrapper scripts to no longer hard-code the absolute path to the installation directory.

### CODE REVIEW

File: haskell/ghc_bindist.bzl [hunk 1]
```diff
@@ -212,8 +212,20 @@ def _ghc_bindist_impl(ctx):
 
     # On Windows the bindist already contains the built executables
     if os != "windows":
-        _execute_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath])
+        _execute_fail_loudly(ctx, ["sed", "-i", "s/RelocatableBuild = NO/RelocatableBuild = YES/", "mk/config.mk.in"])
+        _execute_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath, "--libdir", "${exec_prefix}/lib"])
         _execute_fail_loudly(ctx, ["make", "install"])
+        ctx.file("patch_bins", executable=True, content="""#!/usr/bin/env bash
+set -x
```
Comment (mboes) [inline haskell/ghc_bindist.bzl]: The tracing stuff should probably go.

File: haskell/ghc_bindist.bzl [hunk 2]
```diff
@@ -212,8 +212,20 @@ def _ghc_bindist_impl(ctx):
 
     # On Windows the bindist already contains the built executables
     if os != "windows":
-        _execute_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath])
+        _execute_fail_loudly(ctx, ["sed", "-i", "s/RelocatableBuild = NO/RelocatableBuild = YES/", "mk/config.mk.in"])
+        _execute_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath, "--libdir", "${exec_prefix}/lib"])
```
Comment (mboes) [inline haskell/ghc_bindist.bzl]: Where does `${exec_prefix}` come from?

File: haskell/ghc_bindist.bzl (hunk 1 above)
Comment (aherrmann) [reply haskell/ghc_bindist.bzl]: Removed. Sorry, overlooked that.

File: haskell/ghc_bindist.bzl (hunk 2 above)
Comment (aherrmann) [reply haskell/ghc_bindist.bzl]: `${exec_prefix}` is defined within `mk/install.mk`. However, it turns out that `${exec_prefix}/lib` is the default for `RelocatableBuild = YES`. So the `--libdir` flag is redundant. I've removed it.

**Pull request merged**